client/landing/gutenboarding/stores/onboard/persist.ts

```diff
@@ -0,0 +1,79 @@
+/*
+    Defines the options used for the @wp/data persistence plugin, 
+    which include a persistent storage implementation to add data expiration handling.
+*/
+
+const PERSISTENCE_INTERVAL = 7 * 24 * 3600000; // days * hours in days * ms in hour
+const STORAGE_KEY = 'WP_ONBOARD';
+const STORAGE_TS_KEY = 'WP_ONBOARD_TS';
+
+// A plain object fallback if localStorage is not available
+const objStore: { [ key: string ]: string } = {};
+
+const objStorage: Pick< Storage, 'getItem' | 'setItem' | 'removeItem' > = {
+	getItem( key ) {
+		if ( objStore.hasOwnProperty( key ) ) {
+			return objStore[ key ];
+		}
+
+		return null;
+	},
+	setItem( key, value ) {
+		objStore[ key ] = String( value );
+	},
+	removeItem( key ) {
+		delete objStore[ key ];
+	},
+};
+
+// Make sure localStorage support exists
+const localStorageSupport = (): boolean => {
+	try {
+		window.localStorage.setItem( 'WP_ONBOARD_TEST', '1' );
+		window.localStorage.removeItem( 'WP_ONBOARD_TEST' );
+		return true;
+	} catch ( e ) {
+		return false;
+	}
+};
+
+// Choose the right storage implementation
+const storageHandler = localStorageSupport() ? window.localStorage : objStorage;
+
+// Persisted data expires after seven days
+const isFresh = ( timestampStr: string ): boolean => {
```

Comment:
Small nitpick on naming.  `isFresh` seems a bit more ambiguous than `isExpired` or `isNotExpired`.  `isFresh` makes me think we are loading on a fresh state, similar to when the `?fresh` param is used to clear the data.  But here `isFresh` is `true` when the user comes back after 4 days to pick up where they left off.